lib/open-amp/vendor_rsc_table/nxp_resource_table.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * NXP resource table
+ *
+ * Remote processors include a "resource table" section in addition
+ * to standard ELF segments.
+ *
+ * The resource table lists system resources needed before powering on,
+ * like contiguous memory allocation or IOMMU mapping.
+ * It also includes entries for supported features and configurations,
+ * such as trace buffers and virtio devices.
+ *
+ * Dependencies:
+ *   Must be linked in the ".resource_table" section to comply with
+ *   Linux kernel OS.
+ *
+ * Related documentation:
+ *   https://www.kernel.org/doc/Documentation/remoteproc.txt
+ *   https://openamp.readthedocs.io/en/latest/protocol_details/lifecyclemgmt.html
+ */
+
+#include <zephyr/kernel.h>
+#include <nxp_resource_table.h>
+
+#define __resource Z_GENERIC_SECTION(.resource_table)
+
+static struct nxp_resource_table __resource nxp_rsc_table = {
+	.rsc_table = {
+		#include "../rsc_table_init.h"
+	},
+	.imx_rsc_table = {
+		.len = 0x10,
+		.magic_num = 0x6E787073,
+		.version = 0x0,
+		.features = 0x1,
+	},
```

Comment:
Seems that you need more that a vendor specific type, so you have a specific resource table. 
redefining the `fw_resource_table` seems making sense instead of defining a nxp_resource_table type

Reply:
Redefining the whole `fw_resource_table` is a lot of code duplication. Also, if something changes in the base resource table the vendor specific needs to be updated, it will be hard to maintain.
All that I add in the resource table is just a new vendor specific entry. Other vendors can come with their own entry on top of the base resource table from `rsc_table_init.h`